### PR TITLE
chore(github): add action for release

### PR DESCRIPTION
**Pull request by Francois-Esquire**
Replacing Travis with GitHub action release workflow.

### CODE REVIEW

**Comment by mtomcal** [comment]
Have we run a test of a release in a fork? Not sure how hard that is with lerna.

**Comment by Francois-Esquire** [comment]
> Have we run a test of a release in a fork? Not sure how hard that is with lerna.

No, not yet. Let's hold off merging until we validate this is working as expected.

**Comment by JamesSingleton** [comment]
@Francois-Esquire just an FYI, we will need to add a test action as well since we are deleting `.travis.yml`. By default Travis runs `npm test` on NodeJS.

**Comment by JamesSingleton** [comment]
@Francois-Esquire should we be adding https://github.com/marketplace/actions/create-a-release as well?

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+      - run: NODE_ENV=development npm ci
+      - run: npm run test
+      - run: npm run build
+      - name: Lerna release
+        env:
+          GIT_AUTHOR_EMAIL: ${{ secrets.GIT_AUTHOR_EMAIL }}
+          GIT_AUTHOR_NAME: ${{ secrets.GIT_AUTHOR_NAME }}
+          GIT_COMMITTER_EMAIL: ${{ secrets.GIT_COMMITTER_EMAIL }}
+          GIT_COMMITTER_NAME: ${{ secrets.GIT_COMMITTER_NAME }}
+          GITHUB_TOKEN: ${{ secrets.PA_TOKEN }}
```
Comment (nellyk) [inline .github/workflows/release.yml:35]:
```suggestion
```
This are only needed for semantic release

**Comment by Francois-Esquire** [comment]
> @Francois-Esquire should we be adding https://github.com/marketplace/actions/create-a-release as well? The reason I ask is because if you go to https://github.com/americanexpress/holocron/releases, they do not look like https://github.com/americanexpress/one-app/releases

@JamesSingleton we should look into that

**Review by JAdshead — DISMISSED** [review]

**Review by mtomcal — DISMISSED** [review]

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+          yarn lerna bootstrap
+      - name: Test packages
+        run: yarn test
+      - name: Release packages
+        env:
+          NODE_ENV: production
+          NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
+        run: |-
+          echo "//registry.npmjs.org/:_authToken=\${NPM_TOKEN}" >> $HOME/.npmrc 2> /dev/null
+          yarn lerna:deploy -- "${{ needs.checkCommit.outputs.commitMsg }}"
```
Comment (JamesSingleton) [inline .github/workflows/release.yml:44]: `needs.<job id>.outputs.<output name>` so this looks correct

**Review by JamesSingleton — APPROVED** [review]

**Review by mtomcal — APPROVED** [review]

**Pull request merged**